lib/dns.js

```diff
@@ -133,16 +134,48 @@ function onlookupall(err, addresses) {
   }
 }
 
+function cleanDnsCache() {
+  for (const key in dnsCache) {
+    delete dnsCache[key];
+  }
+}
+
+function scheduleCleanup() {
```

Comment:
Rather than using a timer for cleanup, a lazy check that occurs when the cache is checked would be better. That is, when the cache is checked for a particular key, or when `addDnsEntry` is called, purge expired entries immediately before lookup.

Reply:
This commit replaces timer based purging with lazy loading [090a1a0](https://github.com/nodejs/node/pull/49560/commits/090a1a094d9fa2e65a7b23d4f5cd5df4ddb2b35c), I hope this is what you meant.